miden-lib/asm/sat/tx.masm

```diff
@@ -42,3 +45,60 @@ end
 export.get_output_notes_hash
     exec.epilogue::process_created_notes
 end
+
+#! Increments the number of created notes by one.
+#!
+#! Inputs: []
+#! Outputs: []
+proc.increment_num_created_notes
+    exec.layout::get_num_created_notes
+    add.1
+    exec.layout::set_num_created_notes
+end
```

Comment:
Would it make sense to have this as an atomic operation in the `layout` module?

Reply:
From a performance perspective I'm not sure if it will make a difference as the `get` / `set` proecdures will be inlined anyway. I would propose that we add the note limit assertions into this procedure but leave the procedure in `tx.masm` as it contains validition logic which is outside the scope of `layout`.  I'm not strongly oppinionated so happy to change if you think moving to `layout` is better.

Reply:
Agreed! Let's just add the assertion here.

Reply:
done